sdk/core/azure-core-management/src/test/java/com/azure/core/management/PagedListTests.java

```diff
@@ -252,7 +255,8 @@ public String nextPageLink() {
                         }
                     };
                 }
-                throw new RuntimeException("nextPage should not be called after a page with next link as null");
+                logger.logAndThrow(new RuntimeException("nextPage should not be called after a page with next link as null"));
+                return null;
```

Comment:
This rule doesn't need to be enforced in test classes - these changes should be reverted.

Reply:
What about the changes in the Sample classes?